tests/core_tests.py

```diff
@@ -148,9 +148,22 @@ def test_old_slice_json_endpoint(self):
     def test_slice_json_endpoint(self):
         self.login(username='admin')
         slc = self.get_slice('Girls', db.session)
-        resp = self.get_resp(slc.explore_json_url)
+        # add mock data to trigger POST method
+        resp = self.get_resp(slc.explore_json_url, data={'test': 1})
         assert '"Jennifer"' in resp
 
+    def test_explore_json_endpoint(self):
+        self.login(username='admin')
+        slc = self.get_slice('Girls', db.session)
+
+        # by default, feature CLIENT_CACHE is disabled,
+        # explore_json should not allow GET method
+        resp = self.client.get(slc.explore_json_url)
```

Comment:
@graceguo-supercat maybe it would be worthwhile to have a test case with `CLIENT_CACHE` enabled and one where it's explicitly disabled to the `GET` method returns a `200` and `405` status code respectively.

Reply:
I added another case, so now we test enabled csrf protection flag (GET and POST), and disable csrf protection flag (GET and POST).